docs/delta-lake-migration.md

```diff
@@ -0,0 +1,128 @@
+
+## Compatibilities
+The module is built and tested with `Delta Standalone:0.6.0` and supports Delta Lake tables with the following protocol version:
+* `minReaderVersion`: 1
+* `minWriterVersion`: 2
+
+Please refer to [Delta Lake Table Protocol Versioning](https://docs.delta.io/latest/versioning.html) for more details about Delta Lake protocol versions.
+
+For delta lake table contains `TimestampType` columns, please make sure to set table property `read.parquet.vectorization.enabled` to `false` since the vectorized reader doesn't support `INT96` yet.
+Such support is under development in the [PR#6962](https://github.com/apache/iceberg/pull/6962)
+
+## API
+The `iceberg-delta-lake` module provides an interface named `DeltaLakeToIcebergMigrationActionsProvider`, which contains actions that helps converting from Delta Lake to Iceberg.
+The supported actions are:
+* `snapshotDeltaLakeTable`: snapshot an existing Delta Lake table to an Iceberg table
+
+## Default Implementation
+The `iceberg-delta-lake` module also provides a default implementation of the interface which can be accessed by
+```java
+DeltaLakeToIcebergMigrationActionsProvider defaultActions = DeltaLakeToIcebergMigrationActionsProvider.defaultActions()
+```
+
+### `snapshotDeltaLakeTable`
+The action reads the Delta Lake table's most recent snapshot and converts it to a new Iceberg table with the same schema and partitioning in one iceberg transaction.
+The original Delta Lake table remains unchanged.
+
+The newly created table can be changed or written to without affecting the source table, but the snapshot uses the original table's data files.
+Existing data files are added to the Iceberg table's metadata and can be read using a name-to-id mapping created from the original table schema.
+
+When inserts or overwrites run on the snapshot, new files are placed in the snapshot table's location. The location is default to be the same as that
+of the source Delta Lake Table. Users can also specify a different location for the snapshot table.
+
+{{< hint info >}}
+Because tables created by `snapshotDeltaLakeTable` are not the sole owners of their data files, they are prohibited from
+actions like `expire_snapshots` which would physically delete data files. Iceberg deletes, which only effect metadata,
+are still allowed. In addition, any operations which affect the original data files will disrupt the Snapshot's
+integrity. DELETE statements executed against the original Delta Lake table will remove original data files and the
+`snapshotDeltaLakeTable` table will no longer be able to access them.
+{{< /hint >}}
+
+#### Arguments
+The delta table's location is required to be provided when initializing the action.
+
+| Argument Name | Required? | Type | Description |
+|---------------|-----------|------|-------------|
+|`sourceTableLocation` | Yes | String | The location of the source Delta Lake table | 
+
+#### Configurations
```

Comment:
I think we don't need to add these information in table, we can provide javadoc link

Reply:
Thank you for your suggestions. I agree it looks a liitle redundant as the table just repeat the content in the javadoc. However, I still want to includes the basic Input/Output here so that users can understand the minimum requirements to run the action without being re-direct to other pages. Hence, I re-factored the whole thing as a small table which contains two column which include the Input content and link to the configuration API. For other optional configurations, I just put a javadoc link. Do you think it is proper to refactor like this?